src/smf/packet-smf.c

```diff
@@ -861,32 +877,36 @@ static guint get_smf_pdu_len(packet_info* inf, tvbuff_t *tvb, int offset, void *
 }
 
 /* Add a base-64 encoded string to the tree */
-static void smf_proto_add_base64_string(proto_tree *tree, int id, tvbuff_t *tvb,
+static void smf_proto_add_base64_string(proto_tree *tree, packet_info *pinfo, int id, tvbuff_t *tvb,
     int offset, int size)
 {
     char* str;
     gsize len = size;
```

Comment:
This was bad code to begin with.  Why are we setting len=size.  len (the length returned from g_base64_decode_inplace) is totally different from size (the input size).  Leave len uninitialized.

Reply:
If I leave len uninitialized, then how do I set:
str[len] = '\0'
in situations where g_base64_decode_inplace() is not called?

Reply:
You don't call g_base64_decode_inplace() when strlen(str) < 1.  If str is not already null terminated then you have a bug in that line.

Reply:
Agreed. Removing...